src/test/java/utils/DataUtil.java

```diff
@@ -0,0 +1,154 @@
+package utils;
+
+import org.testng.annotations.DataProvider;
+
+public class DataUtil {
+  @DataProvider(name = "createDataForSubLong")
```

Comment:
Why your approach with data providers is not very effective?

Let's think if there will be 100 methods in the calculator module. So then you are going to have 100 data providers?